addons/mass_mailing/wizard/mailing_mailing_test.py

```diff
@@ -48,10 +57,30 @@ def send_mail_test(self):
                 'notification': True,
                 'mailing_id': mailing.id,
                 'attachment_ids': [(4, attachment.id) for attachment in mailing.attachment_ids],
-                'auto_delete': True,
+                'auto_delete': False,  # they are manually deleted after notifying the document
                 'mail_server_id': mailing.mail_server_id.id,
             }
             mail = self.env['mail.mail'].sudo().create(mail_values)
             mails_sudo |= mail
         mails_sudo.send()
+
+        for mail_sudo in mails_sudo:
+            if mail_sudo.state == 'sent':
+                notification_messages.append(
+                    _('Test mailing successfully sent to %s', mail_sudo.email_to))
+            elif mail_sudo.state == 'exception':
+                notification_messages.append(
+                    _('Test mailing could not be sent to %s:<br>%s',
```

Comment:
Same here, would probably have a list of "failed_recipients", then build the final message to post at the end.

Reply:
The failure reason could be different every time so I guess having one line per mailing is best.
Unless you want me to create a dictionary to group the mailing per error reason to only log once per error but that would be over-engineered for something that is most likely only going to be used on one mailing at a time IMO.